src/pages/index.js

```diff
@@ -54,55 +73,177 @@ const profileEditModal = new ModalWithForm(
+    });
+}
+
 /*    Functions for opening/closing modals    */
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo({ name: data.name, description: data.description });
-  profileEditModal.close();
+  profileEditModal.setModalLoad(true);
+  api
+    .updateUserInfo(data.name, data.description)
+    .then(() => {
+      // Update the UI with the new user data
+      userInfo.setUserInfo(data);
+      editFormValidator.disableButton();
+      profileEditModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      profileEditModal.setModalLoad(false);
+    });
 }
 
 function handleAddCardCreate(data) {
   const name = data.title;
   const link = data.link;
+  addCardModal.setModalLoad(true);
+  api
+    .addCard(name, link)
+    .then((cardData) => {
+      renderCard(cardData);
+      addFormValidator.disableButton();
+      addCardForm.reset();
+      addCardModal.close();
+    })
+    .catch((err) => {
+      console.error("Error adding card:", err);
+    })
+    .finally(() => {
+      addCardModal.setModalLoad(false);
+    });
+}
 
-  console.log(`Creating card with name: ${name} and link: ${link}`);
+const deleteConfirmationModal = new ModalWithConfirmation({
+  modalSelector: "#delete-confirmation-modal",
+});
 
-  renderCard({ name, link });
-  addCardModal.close();
-  addCardForm.reset();
-  addFormValidator.disableButton();
+deleteConfirmationModal.setEventListeners();
+
+function handleDeleteCard(card) {
+  deleteConfirmationModal.open();
+  deleteConfirmationModal.setSubmitFunction(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.handleRemoveCard();
+        deleteConfirmationModal.close();
+      })
+      .catch((err) => {
+        console.error("Error deleting card:", err);
+      });
+  });
+}
+
+function handleImageClick(name, link) {
+  previewImageModal.open({ name, link });
+}
+
+function handleLikeCard(card) {
+  if (card._isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then((data) => {
+        card.handleLikeIcon(false);
+      })
+      .catch((err) => {
+        console.error("Error disliking card:", err);
+      });
+  } else {
+    api
+      .likeCard(card._id)
+      .then((data) => {
+        card.handleLikeIcon(true);
+      })
+      .catch((err) => {
+        console.error("Error liking card:", err);
+      });
+  }
 }
 
 /*    Event Listeners    */
 profileEditButton.addEventListener("click", () => {
+  profileEditModal.open();
   const userData = userInfo.getUserInfo();
   profileNameInput.value = userData.name.trim();
-  profileDescriptionInput.value = userData.job.trim();
+  profileDescriptionInput.value = userData.job.trim(); // Update this line
   editFormValidator.resetValidation();
-  profileEditModal.open();
 });
 
 addCardButton.addEventListener("click", () => {
-  addFormValidator.resetValidation();
   addCardModal.open();
+  addFormValidator.resetValidation();
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. 


You don't need it here